src/library/dao/trackdao.cpp

```diff
@@ -37,16 +36,14 @@ QMutex TrackDAO::m_sTracksMutex;
 
 enum { UndefinedRecordIndex = -2 };
 
-TrackCacheItem::TrackCacheItem(TrackPointer pTrack)
-        : m_pTrack(pTrack) {
-    DEBUG_ASSERT(m_pTrack);
+RecentTrackCacheItem::RecentTrackCacheItem(TrackPointer pTrack)
+        : m_pTrack(std::move(pTrack)) {
```

Comment:
QSharedPointer has no move constructor, so std::move() is redundant. Instead we can pass the pTrack by const reference.